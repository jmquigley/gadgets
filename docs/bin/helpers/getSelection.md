<a name="getSelectionShim"></a>

## getSelectionShim()
DOM Traversal is not implemented in JSDOM
The best we can do is shim the functions

**Kind**: global function  
