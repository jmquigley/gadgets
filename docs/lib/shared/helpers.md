## Functions

<dl>
<dt><a href="#globalize">globalize(name, pkg, replace)</a> ⇒ <code>object</code></dt>
<dd><p>Takes a variable name and an object and places that name and associated
object into the global node space.  If the object is already in the
global space, then that reference is used and returned.  The replace
flag is used to override the current global reference.</p>
</dd>
<dt><a href="#isDebug">isDebug()</a> ⇒</dt>
<dd><p>Checks the environment for the debug flag environment variable.  If it is
defined, then it will resolve to true.  If it doesn&#39;t, exists, then the
package.json debug flag is checked.  If it exists it is returned as the
debug flag.  If it doesn&#39;t exist, then false is returned and debugging
is turned off.</p>
<p>The name of the environment variable for debugging is DEBUG_GADGETS if
in a node environment.</p>
</dd>
</dl>

<a name="globalize"></a>

## globalize(name, pkg, replace) ⇒ <code>object</code>
Takes a variable name and an object and places that name and associated
object into the global node space.  If the object is already in the
global space, then that reference is used and returned.  The replace
flag is used to override the current global reference.

**Kind**: global function  
**Returns**: <code>object</code> - the global instance reference  
**Params**

- name <code>string</code> - the name of the variable to store globally
- pkg <code>object</code> - the object associated with the variable name
- replace <code>boolean</code> - if true, then it will always overwrite the
current global (if it exists)

<a name="isDebug"></a>

## isDebug() ⇒
Checks the environment for the debug flag environment variable.  If it is
defined, then it will resolve to true.  If it doesn't, exists, then the
package.json debug flag is checked.  If it exists it is returned as the
debug flag.  If it doesn't exist, then false is returned and debugging
is turned off.

The name of the environment variable for debugging is DEBUG_GADGETS if
in a node environment.

**Kind**: global function  
**Returns**: true if debugging is turned on, otherwise false.  
