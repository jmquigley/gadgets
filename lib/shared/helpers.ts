import {isNode} from "util.toolbox";

const debug = require("debug")("gadgets.shared.helpers");
const pkg = require("../../package.json");

/**
 * Takes a variable name and an object and places that name and associated
 * object into the global node space.  If the object is already in the
 * global space, then that reference is used and returned.  The replace
 * flag is used to override the current global reference.
 * @param name {string} the name of the variable to store globally
 * @param pkg {object} the object associated with the variable name
 * @param replace {boolean} if true, then it will always overwrite the
 * current global (if it exists)
 * @return {object} the global instance reference
 */
export function globalize(
	name: string,
	pkgname: any,
	replace: boolean = false
) {
	let ref: any = pkgname;
	if (!(global as any)[name] || replace) {
		(window as any)[name] = (global as any)[name] = pkgname;
		debug("Adding global variable %o -> %O", name, (global as any)[name]);
	} else {
		ref = (global as any)[name];
	}

	return ref;
}

/**
 * Checks the environment for the debug flag environment variable.  If it is
 * defined, then it will resolve to true.  If it doesn't, exists, then the
 * package.json debug flag is checked.  If it exists it is returned as the
 * debug flag.  If it doesn't exist, then false is returned and debugging
 * is turned off.
 *
 * The name of the environment variable for debugging is DEBUG_GADGETS if
 * in a node environment.
 * @return true if debugging is turned on, otherwise false.
 */
export function isDebug(): boolean {
	if (isNode() && "DEBUG_GADGETS" in process.env) {
		return true;
	}

	if (pkg.debug) {
		return pkg.debug;
	}

	return false;
}
