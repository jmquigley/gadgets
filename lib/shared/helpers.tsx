'use strict';

const debugEntry = require('debug');
const pkg = require('../../package.json');

/**
 * A wrapper for the debug function.  It uses the package.json to enable or
 * disable debugging information in the library.  It also turns off
 * coverage checks for this function.
 * @param context {string} the name of the module where this debug request
 * is made.
 * @param args {object[]} the list of arguments passed to the the debug
 * function.
 */
export function debug(context: string, ...args: any[]) {
	/* istanbul ignore if  */
	if (pkg.debug) {
		const debugFn = debugEntry(`Gadgets -> ${context}`);
		debugFn(...args);
	}
}

/**
 * Takes a variable name and an object and places that name and associated
 * object into the global node space.  If the object is alread in the
 * global space, then that reference is used and returned.  The replace
 * flag is used to override the current global reference.
 * @param name {string} the name of the variable to store globally
 * @param pkg {object} the object associated with the variable name
 * @param replace {boolean} if true, then it will always overwrite the
 * current global (if it exists)
 * @return {object} the global instance reference
 */
export function globalize(name: string, pkgname: any, replace: boolean = false) {
	let ref: any = pkgname;
	if (!(global as any)[name] || replace) {
		(window as any)[name] = (global as any)[name] = pkgname;
	} else {
		ref = (global as any)[name];
	}

	return ref;
}
