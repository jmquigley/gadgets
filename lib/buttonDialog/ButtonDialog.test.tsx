'use strict';

import {cleanup, log, mockupEnv} from '../../test/helpers';
mockupEnv();

import test from 'ava';
import {mount} from 'enzyme';
import * as path from 'path';
import * as React from 'react';
import {ButtonDialog} from './index';

test.after.always.cb(t => {
	cleanup(path.basename(__filename), t);
});

test('Test creation of a ButtonDialog control', t => {
	const ctl = mount(
		<ButtonDialog className="test-class">
			<p>Dialog test</p>
		</ButtonDialog>
	);

	t.truthy(ctl);
	log.debug(ctl.html(), __filename);

	t.is(ctl.prop('iconName'), 'bomb');
	t.false(ctl.prop('disabled'));
	t.true(ctl.prop('visible'));
	t.is(ctl.prop('id'), undefined);
	t.true(ctl.contains(<p>Dialog test</p>));
	t.is(ctl.find('.test-class').length, 1);
});

test('Test the click event on a ButtonDialog control', t => {
	const ctl = mount(
		<ButtonDialog>
			<p>Dialog test</p>
		</ButtonDialog>
	);

	t.truthy(ctl);
	log.debug(ctl.html(), __filename);

	t.is(ctl.prop('iconName'), 'bomb');
	t.false(ctl.prop('disabled'));
	t.true(ctl.prop('visible'));
	t.is(ctl.prop('id'), undefined);
	t.true(ctl.contains(<p>Dialog test</p>));

	ctl.find('i').simulate('click');
	t.true(ctl.state('visible'));
});

test('Test the disabling of the ButtonDialog control', t => {
	const ctl = mount(
		<ButtonDialog disabled={true}>
			<p>Dialog test</p>
		</ButtonDialog>
	);

	t.truthy(ctl);
	log.debug(ctl.html(), __filename);

	t.is(ctl.prop('iconName'), 'bomb');
	t.true(ctl.prop('disabled'));
	t.true(ctl.prop('visible'));
	t.is(ctl.prop('id'), undefined);
	t.is(ctl.find('.disabled').length, 2); // on control and button

	ctl.find('i').simulate('click');
	t.false(ctl.state('visible'));
});

test('Test the making the ButtonDialog invisible', t => {
	const ctl = mount(
		<ButtonDialog visible={false}>
			<p>Dialog test</p>
		</ButtonDialog>
	);

	t.truthy(ctl);
	log.debug(ctl.html(), __filename);

	t.is(ctl.prop('iconName'), 'bomb');
	t.false(ctl.prop('disabled'));
	t.false(ctl.prop('visible'));
	t.is(ctl.prop('id'), undefined);
	t.is(ctl.find('.invisible').length, 2);
	t.true(ctl.contains(<p>Dialog test</p>));

	ctl.find('i').simulate('click');
	t.false(ctl.state('visible'));
});

// :TODO: Add tests for exposed dialog popup
