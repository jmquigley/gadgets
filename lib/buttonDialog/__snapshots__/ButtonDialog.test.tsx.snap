// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test creation of a ButtonDialog control 1`] = `
<Wrapper
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  onError={[Function]}
  padding={null}
  reset={false}
  ripple={false}
  selected={false}
  sizing="normal"
  style={
    Object {
      "whiteSpace": "pre-wrap",
    }
  }
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
>
  <ButtonDialog__ButtonDialogView
    className="ui-button-dialog test-class ripple"
    id="ButtonDialog-0"
  >
    <Button
      backgroundColor={null}
      border={null}
      borderColor={null}
      bottom={null}
      className={null}
      color={null}
      controlled={true}
      direction="right"
      disabled={false}
      err={null}
      errorMessage={null}
      focus={false}
      height={null}
      hidden={false}
      iconName="bomb"
      iconStyle=""
      id={null}
      kbActivate=""
      left={null}
      location="none"
      maxHeight={null}
      maxWidth={null}
      minHeight={null}
      minWidth={null}
      noborder={false}
      noedit={false}
      nohover={false}
      nopropagation={true}
      notheme={false}
      notooltip={false}
      onClick={[Function]}
      padding={null}
      ripple={true}
      selected={false}
      sizing="normal"
      style={
        Object {
          "backgroundColor": "inherit",
          "color": "inherit",
        }
      }
      testing={true}
      theme={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#c0c0c0",
          "buttonColor": "#666666",
          "chevronColor": "#cccccc",
          "color": "#000000",
          "fontFamily": "OpenSans, Georgia, serif",
          "headerBackgroundColor": "#004358",
          "headerForegroundColor": "#ffffff",
          "headerHoverColor": "#ffe11a",
          "hoverColor": "#c0c0c0",
          "inputBorderColor": "#c0c0c0",
          "itemHoverColor": "#bedb39",
          "outlineColor": "#51cbee",
          "searchFocus": "#51cbee",
          "searchMatch": "#fd7400",
          "selectedBackgroundColor": "#1f8a70",
          "selectedForegroundColor": "#ffe11a",
          "titleBarBackgroundColor": "#004358",
          "titleBarForegroundColor": "#ffe11a",
          "tooltipBackgroundColor": "#708090",
          "tooltipForegroundColor": "#ffffff",
          "transitionDelay": "0.33s",
        }
      }
      tooltip={null}
      top={null}
      visible={true}
      width={null}
      xcss={null}
    />
    <ButtonDialog__ButtonDialogPopup
      className="ui-dialog-popup test-class-dialog"
      location="bottom"
      onClick={[Function]}
      visible={false}
    >
      <ButtonDialog__ButtonDialogContent
        sizing="normal"
        visible={false}
      >
        <p>
          Dialog test
        </p>
      </ButtonDialog__ButtonDialogContent>
      <ButtonDialog__StyledTriangle
        className="ui-dialog-triangle"
        direction="up"
        location="bottom"
        nobase={true}
        sizing="normal"
        style={
          Object {
            "fill": "#ffffff",
            "stroke": "#c0c0c0",
          }
        }
        visible={false}
      />
    </ButtonDialog__ButtonDialogPopup>
  </ButtonDialog__ButtonDialogView>
</Wrapper>
`;

exports[`Test creation of a ButtonDialog control on top 1`] = `
<Wrapper
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  id={null}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  onError={[Function]}
  padding={null}
  reset={false}
  ripple={false}
  selected={false}
  sizing="normal"
  style={
    Object {
      "whiteSpace": "pre-wrap",
    }
  }
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
>
  <ButtonDialog__ButtonDialogView
    className="ui-button-dialog test-class ripple"
    id="ButtonDialog-0"
  >
    <Button
      backgroundColor={null}
      border={null}
      borderColor={null}
      bottom={null}
      className={null}
      color={null}
      controlled={true}
      direction="right"
      disabled={false}
      err={null}
      errorMessage={null}
      focus={false}
      height={null}
      hidden={false}
      iconName="bomb"
      iconStyle=""
      id={null}
      kbActivate=""
      left={null}
      location="none"
      maxHeight={null}
      maxWidth={null}
      minHeight={null}
      minWidth={null}
      noborder={false}
      noedit={false}
      nohover={false}
      nopropagation={true}
      notheme={false}
      notooltip={false}
      onClick={[Function]}
      padding={null}
      ripple={true}
      selected={false}
      sizing="normal"
      style={
        Object {
          "backgroundColor": "inherit",
          "color": "inherit",
        }
      }
      testing={true}
      theme={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#c0c0c0",
          "buttonColor": "#666666",
          "chevronColor": "#cccccc",
          "color": "#000000",
          "fontFamily": "OpenSans, Georgia, serif",
          "headerBackgroundColor": "#004358",
          "headerForegroundColor": "#ffffff",
          "headerHoverColor": "#ffe11a",
          "hoverColor": "#c0c0c0",
          "inputBorderColor": "#c0c0c0",
          "itemHoverColor": "#bedb39",
          "outlineColor": "#51cbee",
          "searchFocus": "#51cbee",
          "searchMatch": "#fd7400",
          "selectedBackgroundColor": "#1f8a70",
          "selectedForegroundColor": "#ffe11a",
          "titleBarBackgroundColor": "#004358",
          "titleBarForegroundColor": "#ffe11a",
          "tooltipBackgroundColor": "#708090",
          "tooltipForegroundColor": "#ffffff",
          "transitionDelay": "0.33s",
        }
      }
      tooltip={null}
      top={null}
      visible={true}
      width={null}
      xcss={null}
    />
    <ButtonDialog__ButtonDialogPopup
      className="ui-dialog-popup test-class-dialog"
      location="top"
      onClick={[Function]}
      visible={false}
    >
      <ButtonDialog__ButtonDialogContent
        sizing="normal"
        visible={false}
      >
        <p>
          Dialog test
        </p>
      </ButtonDialog__ButtonDialogContent>
      <ButtonDialog__StyledTriangle
        className="ui-dialog-triangle"
        direction="down"
        location="top"
        nobase={true}
        sizing="normal"
        style={
          Object {
            "fill": "#ffffff",
            "stroke": "#c0c0c0",
          }
        }
        visible={false}
      />
    </ButtonDialog__ButtonDialogPopup>
  </ButtonDialog__ButtonDialogView>
</Wrapper>
`;

exports[`Test making the ButtonDialog hidden 1`] = `
<ButtonDialog
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  controlled={true}
  dialogClasses={Array []}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={true}
  iconName="bomb"
  iconStyle=""
  id={null}
  kbActivate=""
  left={null}
  location="bottom"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={true}
  notheme={false}
  notooltip={false}
  notriangle={false}
  onClick={[Function]}
  padding={null}
  ripple={true}
  selected={false}
  sizing="normal"
  style={
    Object {
      "backgroundColor": "inherit",
      "color": "inherit",
    }
  }
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  triangleClasses={Array []}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Test retrieval of ButtonDialog props object 1`] = `
Object {
  "backgroundColor": null,
  "border": null,
  "borderColor": null,
  "bottom": null,
  "children": null,
  "className": null,
  "color": null,
  "controlled": true,
  "dialogClasses": Array [],
  "direction": "right",
  "disabled": false,
  "err": null,
  "errorMessage": null,
  "focus": false,
  "height": null,
  "hidden": false,
  "iconName": "bomb",
  "iconStyle": "",
  "id": null,
  "kbActivate": "",
  "left": null,
  "location": "bottom",
  "maxHeight": null,
  "maxWidth": null,
  "minHeight": null,
  "minWidth": null,
  "noborder": false,
  "noedit": false,
  "nohover": false,
  "nopropagation": true,
  "notheme": false,
  "notooltip": false,
  "notriangle": false,
  "onClick": [Function],
  "padding": null,
  "ripple": true,
  "selected": false,
  "sizing": "normal",
  "style": Object {
    "backgroundColor": "inherit",
    "color": "inherit",
  },
  "testing": true,
  "theme": Object {
    "backgroundColor": "#ffffff",
    "borderColor": "#c0c0c0",
    "buttonColor": "#666666",
    "chevronColor": "#cccccc",
    "color": "#000000",
    "fontFamily": "OpenSans, Georgia, serif",
    "headerBackgroundColor": "#004358",
    "headerForegroundColor": "#ffffff",
    "headerHoverColor": "#ffe11a",
    "hoverColor": "#c0c0c0",
    "inputBorderColor": "#c0c0c0",
    "itemHoverColor": "#bedb39",
    "outlineColor": "#51cbee",
    "searchFocus": "#51cbee",
    "searchMatch": "#fd7400",
    "selectedBackgroundColor": "#1f8a70",
    "selectedForegroundColor": "#ffe11a",
    "titleBarBackgroundColor": "#004358",
    "titleBarForegroundColor": "#ffe11a",
    "tooltipBackgroundColor": "#708090",
    "tooltipForegroundColor": "#ffffff",
    "transitionDelay": "0.33s",
  },
  "tooltip": null,
  "top": null,
  "triangleClasses": Array [],
  "visible": true,
  "width": null,
  "xcss": null,
}
`;

exports[`Test retrieval of ButtonDialog props object 2`] = `
Object {
  "children": null,
  "sizing": "normal",
  "style": Object {},
  "visible": false,
}
`;

exports[`Test the disabling of the ButtonDialog control 1`] = `
<ButtonDialog
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  controlled={true}
  dialogClasses={Array []}
  direction="right"
  disabled={true}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  iconName="bomb"
  iconStyle=""
  id={null}
  kbActivate=""
  left={null}
  location="bottom"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={true}
  notheme={false}
  notooltip={false}
  notriangle={false}
  onClick={[Function]}
  padding={null}
  ripple={true}
  selected={false}
  sizing="normal"
  style={
    Object {
      "backgroundColor": "inherit",
      "color": "inherit",
    }
  }
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  triangleClasses={Array []}
  visible={true}
  width={null}
  xcss={null}
/>
`;
